File: course_inf.cpp
#include "course_inf.h"
#include <iostream>
#include <string>
#include <vector>
#include <cstdio> // For remove() function
//#include <windows.h> // For WinAPI


using namespace std;


// Constructor definition
course_inf::course_inf() {
    vector<assignment_inf*> vec; // added by rx
    assignments = vec; // added by rx
}

// Public member function definition to input course information
void course_inf::inputInfo() {
    cout << "Enter course name: ";
    getline(cin, courseName);

    cout << "Enter grade weight: ";
    cin >> gradeWeight;

    // Calculate individual weights based on overall grade weight
    examWeight = gradeWeight * 0.4; // 40% for exams
    homeworkWeight = gradeWeight * 0.3; // 30% for homeworks
    assignmentWeight = gradeWeight * 0.2; // 20% for assignments
    quizWeight = gradeWeight * 0.1; // 10% for quizzes
}

// Public member function definition to input assignment data
void course_inf::inputAssignment() {
    string assignmentName;
    string assignmentType;
    double assignmentGrade;
    assignment_inf* assignment;

    cin.ignore(); // Clear the newline character left in the input buffer

    cout << "Enter assignment name: ";
    getline(cin, assignmentName);

    cout << "Enter assignment type: ";
    getline(cin, assignmentType);

    cout << "Enter assignment grade: ";
    cin >> assignmentGrade;

    assignment_inf* a = new assignment_inf(assignmentName, assignmentType, assignmentGrade); // added by rx
    assignments.push_back(a); // added by rx

    ofstream outFile("assignment_data.txt", ios::app); // Append mode

    if (outFile.is_open()) {
        outFile << "Assignment Name: " << assignmentName << ", Type: " << assignmentType << ", Grade: " << assignmentGrade << endl;
        outFile.close();
        cout << "Assignment data has been saved to assignment_data.txt" << endl;
    }
    else {
        cerr << "Unable to open file for writing: course_data.txt" << endl;
    }
}

// Public member function definition to display course information
void course_inf::displayInfo() const {
    cout << "Course Name: " << courseName << endl;
    cout << "Grade Weight: " << gradeWeight << endl;
    cout << "Exam Weight: " << examWeight << endl;
    cout << "Homework Weight: " << homeworkWeight << endl;
    cout << "Assignment Weight: " << assignmentWeight << endl;
    cout << "Quiz Weight: " << quizWeight << endl;
}

// Public member function definition to save course information to a file
void course_inf::saveToFile(const string& filename) {
    ofstream outFile(filename);

    if (outFile.is_open()) {
        outFile << "Course Name: " << courseName << endl;
        outFile << "Grade Weight: " << gradeWeight << endl;
        outFile << "Exam Weight: " << examWeight << endl;
        outFile << "Homework Weight: " << homeworkWeight << endl;
        outFile << "Assignment Weight: " << assignmentWeight << endl;
        outFile << "Quiz Weight: " << quizWeight << endl;

        outFile.close();
        cout << "Course information has been saved to " << filename << endl;
    }
    else {
        cerr << "Unable to open file for writing: " << filename << endl;
    }
}

// Public member function definition to read course information from a file
void course_inf::readFromFile(const string& filename) {
    ifstream inFile(filename);

    if (inFile.is_open()) {
        string line;
        while (getline(inFile, line)) {
            cout << line << endl;
        }

        inFile.close();
    }
    else {
        cerr << "Unable to open file for reading: " << filename << endl;
    }
}

/*
void course_inf::deleteAllCourseData() {
    WIN32_FIND_DATAA findFileData;
    HANDLE hFind = FindFirstFileA("*", &findFileData);
    if (hFind != INVALID_HANDLE_VALUE) {
        do {
            if ((findFileData.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY) == 0 &&
                strstr(findFileData.cFileName, ".txt")) {
                std::string filename = findFileData.cFileName;
                std::remove(filename.c_str());
                std::cout << "Deleted file: " << filename << std::endl;
            }
        } while (FindNextFileA(hFind, &findFileData) != 0);
        FindClose(hFind);
    }
    assignments.clear(); // added by rx
}
*/

File: Rita's work/COURSE_INF.py
import os

class AssignmentInf:
    def __init__(self, n="", t="", g=0):
        self.type = t
        self.name = n
        self.grade = g

class CourseInf:
    def __init__(self):
        self.courseName = ""
        self.gradeWeight = 0
        self.examWeight = 0
        self.homeworkWeight = 0
        self.assignmentWeight = 0
        self.quizWeight = 0
        self.assignments = []

    def input_info(self):
        self.courseName = input("Enter course name: ")
        self.gradeWeight = float(input("Enter grade weight: "))
        self.examWeight = self.gradeWeight * 0.4  # 40% for exams
        self.homeworkWeight = self.gradeWeight * 0.3  # 30% for homeworks
        self.assignmentWeight = self.gradeWeight * 0.2  # 20% for assignments
        self.quizWeight = self.gradeWeight * 0.1  # 10% for quizzes

    def input_assignment(self):
        assignment_name = input("Enter assignment name: ")
        assignment_type = input("Enter assignment type: ")
        assignment_grade = float(input("Enter assignment grade: "))
        assignment = AssignmentInf(assignment_name, assignment_type, assignment_grade)
        self.assignments.append(assignment)
        self.save_to_file("assignment_data.txt", assignment_name, assignment_type, assignment_grade)

    def display_info(self):
        print("Course Name:", self.courseName)
        print("Grade Weight:", self.gradeWeight)
        print("Exam Weight:", self.examWeight)
        print("Homework Weight:", self.homeworkWeight)
        print("Assignment Weight:", self.assignmentWeight)
        print("Quiz Weight:", self.quizWeight)

    def save_to_file(self, filename, assignment_name, assignment_type, assignment_grade):
        with open(filename, 'a') as file:
            file.write(f"Assignment Name: {assignment_name}, Type: {assignment_type}, Grade: {assignment_grade}\n")
        print(f"Assignment data has been saved to {filename}")

    def save_course_info_to_file(self, filename):
        with open(filename, 'w') as file:
            file.write(f"Course Name: {self.courseName}\n")
            file.write(f"Grade Weight: {self.gradeWeight}\n")
            file.write(f"Exam Weight: {self.examWeight}\n")
            file.write(f"Homework Weight: {self.homeworkWeight}\n")
            file.write(f"Assignment Weight: {self.assignmentWeight}\n")
            file.write(f"Quiz Weight: {self.quizWeight}\n")
        print(f"Course information has been saved to {filename}")

    def read_from_file(self, filename):
        if os.path.exists(filename):
            with open(filename, 'r') as file:
                for line in file:
                    print(line.strip())
        else:
            print(f"File {filename} does not exist")

    @staticmethod
    def delete_all_course_data():
        for filename in os.listdir():
            if filename.endswith(".txt"):
                os.remove(filename)
                print(f"Deleted file: {filename}")

# Usage Example:
'''
course = CourseInf()
course.input_info()
course.input_assignment()
course.display_info()
course.save_course_info_to_file("course_data.txt")
course.read_from_file("course_data.txt")
CourseInf.delete_all_course_data()
'''
